src/main/java/org/jsuffixarrays/SuffixArrays.java

```diff
@@ -1,170 +1,188 @@
+ * </ul>
+ */
+public final class SuffixArrays
+{
+    /**
+     * Maximum required trailing space in the input array (certain algorithms need it).
+     */
+    final static int MAX_EXTRA_TRAILING_SPACE = DeepShallow.OVERSHOOT;
+
+    /*
+	 * 
+	 */
+    private SuffixArrays()
+    {
+        // no instances.
+    }
+
+    /**
+     * Create a suffix array for a given character sequence with the default algorithm.
+     */
+    public static int [] create(CharSequence s)
+    {
+        return create(s, defaultAlgorithm());
+    }
+
+    /**
+     * Create a suffix array for a given character sequence, using the provided suffix
+     * array building strategy.
+     */
+    public static int [] create(CharSequence s, ISuffixArrayBuilder builder)
+    {
+        return new CharSequenceAdapter(builder).buildSuffixArray(s);
+    }
+
+    /**
+     * Create a suffix array and an LCP array for a given character sequence.
+     * 
+     * @see #computeLCP(int[], int, int, int[])
+     */
+    public static SuffixData createWithLCP(CharSequence s)
+    {
+        return createWithLCP(s, defaultAlgorithm());
+    }
+
+    /**
+     * Create a suffix array and an LCP array for a given character sequence, use the
+     * given algorithm for building the suffix array.
+     * 
+     * @see #computeLCP(int[], int, int, int[])
+     */
+    public static SuffixData createWithLCP(CharSequence s, ISuffixArrayBuilder builder)
+    {
+        final CharSequenceAdapter adapter = new CharSequenceAdapter(builder);
+        final int [] sa = adapter.buildSuffixArray(s);
+        final int [] lcp = computeLCP(adapter.input, 0, s.length(), sa);
+        return new SuffixData(sa, lcp);
+    }
+
+    /**
+     * Create a suffix array and an LCP array for a given input sequence of symbols.
+     */
+    public static SuffixData createWithLCP(int [] input, int start, int length)
+    {
+        final ISuffixArrayBuilder builder = new DensePositiveDecorator(
+            new ExtraTrailingCellsDecorator(defaultAlgorithm(), 3));
+        return createWithLCP(input, start, length, builder);
+    }
+
+    /**
+     * Create a suffix array and an LCP array for a given input sequence of symbols and a
+     * custom suffix array building strategy.
+     */
+    public static SuffixData createWithLCP(int [] input, int start, int length,
+        ISuffixArrayBuilder builder)
+    {
+        final int [] sa = builder.buildSuffixArray(input, start, length);
+        final int [] lcp = computeLCP(input, start, length, sa);
+        return new SuffixData(sa, lcp);
+    }
+
+    public static <T extends Comparable> SuffixData createWithLCP(T[] input, ISuffixArrayBuilder builder) {
+        return createWithLCP(input, builder, null);
+    }
+
+    /**
+     * Create a suffix array and an LCP array for a given generic array and a
+     * custom suffix array building strategy, using the given T object
+     * comparator.
+     */
+    public static <T> SuffixData createWithLCP(T[] input,
```

Comment:
Only here, createWithLCP(T[], ISufficArrayBuilder, Comparator<? super T>),  seems to be added in this class, but the diff shows everything changed in this file. Change spaces?